website/content/en/docs/concepts/scheduling.md

```diff
@@ -260,6 +260,14 @@ If they all fail, Karpenter will fail to provision the pod.
 Karpenter will backoff and retry over time.
 So if capacity becomes available, it will schedule the pod without user intervention.
 
+#### Preferred Affinities
+
+Karpenter treats preferred affinities as hard affinities in almost all circumstances.
+
+Karpenter interprets preferred affinities as a soft affinity when building up topology requirements for scheduling to a node. Preferred affinities are ignored in regards to topology zones and hard affinities should be used as documented above.
```

Comment:
Oh, this is interesting. I think it's a little unclear without an example, but I do get what you are saying: Since TSC consider nodeAffinity when considering which domains to consider, you are calling out that this doesn't cause Karpenter to unnecessarily constrain the TSC when these are listed as preferences. It's definitely bespoke. Maybe worth calling this out as an `{{% alert %}}` since this is more like edge behavior

Reply:
Added a warning alert.